Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift

```diff
@@ -156,21 +161,127 @@ public struct MessageParameter: Encodable {
       // This should be a uuid, hash value, or other opaque identifier. Anthropic may use this id to help detect abuse. Do not include any identifying information such as name, email address, or phone number.
       let userId: UUID
    }
-   
+    
+    public struct Function {
+        let name: String
+        let description: String
+        let parameters: [Parameter]
+  
+        public init(name: String, description: String, parameters: [Parameter]) {
+            self.name = name
+            self.description = description
+            self.parameters = parameters
+        }
+        
+        public struct Parameter {
+            let name: String
+            let type: ParamType
+            let description: String
+            
+            public init(name: String, type: ParamType, description: String) {
+                self.name = name
+                self.type = type
+                self.description = description
+            }
+            
+            public enum ParamType: String {
+                case string
+                case integer
+                case number
+            }
+
+            public func toXML() -> String {
+                return """
+            <parameter>
+                <name>\(name)</name>
+                <type>\(type)</type>
+                <description>\(description)</description>
+            </parameter>
+            """
+            }
+        }
+        
+        public func toXML() -> String {
+            return """
+        <tool_description>
+            <tool_name>\(name)</tool_name>
+            <description>\(description)</description>
+            <parameters>
+                \(parameters.map { $0.toXML() }.joined(separator:"\n\t\t"))
+            </parameters>
+        </tool_description>
+        """
```

Comment:
you can indent this all one more notch. the indentation within the string itself matches the closing `"""` not the opening one or the code block its contained in